modules/calib3d/src/usac/pnp_solver.cpp

```diff
@@ -284,25 +289,30 @@ class P3PSolverImpl : public P3PSolver {
      * calibrated normalized points
      * K^-1 [u v 1]^T / ||K^-1 [u v 1]^T||
      */
-    const Mat * points_mat, * calib_norm_points_mat;
+    const Mat points_mat, calib_norm_points_mat;
     const Matx33d K;
-    const float * const calib_norm_points, * const points;
     const double VAL_THR = 1e-4;
 public:
     /*
      * @points_ is matrix N x 5
      * u v x y z. (u,v) is image point, (x y z) is world point
      */
     P3PSolverImpl (const Mat &points_, const Mat &calib_norm_points_, const Mat &K_) :
-        points_mat(&points_), calib_norm_points_mat(&calib_norm_points_), K(K_),
-        calib_norm_points((float*)calib_norm_points_mat->data), points((float*)points_mat->data) {}
+        points_mat(points_), calib_norm_points_mat(calib_norm_points_), K(K_)
+    {
+        CV_DbgAssert(!points_mat.empty() && points_mat.isContinuous());
+        CV_DbgAssert(!calib_norm_points_mat.empty() && calib_norm_points_mat.isContinuous());
```

Comment:
I'm wondering if it is possible to use `CV_Assert` instead of debug. Is it performance sensitive code?

Reply:
There is continuity check and check for emptiness in interface USAC functions. The checks originally were added for debug purposes. I propose to remain them as debug.